standards/react_native_standards.md

```diff
@@ -0,0 +1,121 @@
+- **State Management**: Keep state as localized as possible to avoid unnecessary re-renders and maintain a clear data flow.
+- **Performance Optimization**: Minimize unnecessary re-renders, efficiently render large lists of data, and profile and optimize performance bottlenecks using react tools.
+- **Platform-Specific Considerations**: Utilize platform-specific components or styles when necessary to accommodate differences between iOS and Android.
+- **Testing and Debugging**: Write comprehensive tests to ensure the reliability and stability of your React Native application.
+
+## Basic Rules
+
+- Always use `const` or `let` to declare variables. Use const by default, unless a variable needs to be reassigned.
+- Always use functional component.
+- No nested components or getComponent inside render method. Always try to separate the components if possible.
+
+## Naming Conventions
+
+- Component’s names should be written using `pascal case`.
+  >`Header.js` <br>
+  >`CustomButton.js` <br>
+  >`Footer.js`
+
+- Non-component’s name should be written using `camel case`.
+  > `myUtilityFile.js` <br>
+  > `cookieHelper.js` <br>
+  > `fetchApi.js`
+
+- Unit test files should use the same name as its corresponding file.
+  > `HeroBanner.js` <br>
+  > `HeroBanner.test.js` <br>
+  
+  > `CookieBanner.js` <br>
+  > `CookieBanner.test.js`
+
+- Global StyleSheet file should be written using `camel case`.
+  > `globalStyle.js` <br>
+
+- All fonts, colors, and font sizes should be in one file.
+  > `fontsConfig.js`
+
+- Attribute name should be `camel case`.
+  > `className` <br>
+  > `onClick`
+
+- Variable names should be `camel case`. Variable names can contain number and special characters.
+  > `const variable = 'test';` <br>
+  > `let variableBoolean = true;`
+
+- Use `.jsx` or `.tsx` extension a for React components
+
+## Files and Folders Structure
+
+###  Grouping by Features or Navigation
+- One of the common approach to structure projects is to locate all main screens in one folder, all reusable components in one folder and style related files in one folder grouped by feature or navigation.
+
+###  Grouping by File Type
+- Another popular approach to structure projects is to group similar files together.
+
+## Importing Modules
+
+###  Import Grouping
+- Standard Libraries (React, React-Native)
+
+  ```csharp
+  import React from 'react';
+  import { TouchableOpacity, View } from 'react-native';
+  ```
+
+- Third-Party Libraries
+
+  ```csharp
+  import axios from 'axios';
+  import lodash from 'lodash';
+  ```
+
+- Your Own Code Imports
+
+  ```csharp
+  import { Button, Card } from '../components';
+  import { MainLayout } from '../layouts';
+  ```
+
+- Module-Specific Imports (styles, PNG, etc.)
+
+  ```csharp
+  import styles from'./myComponent.styles';
+  import image from './image.png';
+  ```
+
+- Test-Only Imports
+
+  ```csharp
+  import { testFunction } from './tests';
+  ```
```

Comment:
Mention the specific eslint-config responsible for this, this should be automated